validator.js

```diff
@@ -450,7 +450,26 @@
     };
 
     validator.isDate = function (str) {
-        return !isNaN(Date.parse(str));
+        var rawDate = new Date(str);
+        var normalizedDate = new Date(rawDate.getTime() + rawDate.getTimezoneOffset()*60000);
+        var dayOrYear, dayOrYearMatches, day, year;
+        if (isNaN(Date.parse(rawDate))) {
+            return false;
+        }
+        dayOrYearMatches = str.match(/[23]\d+/g);
```

Comment:
The intention is to catch all days that are invalid. Unfortunately, years can be caught also since date strings like this are possible:
`'2/22/23'`: Wed Feb 22 2023
`'2-23-22'`: Wed Feb 23 2022

It's hard to tell which is the day or the year for an arbitrary string, so the below code checks for both. It's not part of the test suite, but leap years before 2000 are handled properly.